sources/packages/backend/apps/api/src/route-controllers/audit/_tests_/audit.controller.e2e-spec.ts

```diff
@@ -0,0 +1,124 @@
+      .post(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED);
+
+    expect(auditService.logger.log).toHaveBeenCalledWith(
+      "SIMS Audit Event From ::ffff:127.0.0.1 | User GUID: ministry-user-aest-operations@e2e-tests, Event: Logged In, Portal: Ministry.",
+    );
+  });
+
+  it(`Should log 'Logged Out' message when ${AuditEvent.LoggedOut} is requested.`, async () => {
+    // Arrange
+    const endpoint = `/audit/${AuditEvent.LoggedOut}`;
+    const token = await getInstitutionToken(InstitutionTokenTypes.CollegeCUser);
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED);
+
+    expect(auditService.logger.log).toHaveBeenCalledWith(
+      "SIMS Audit Event From ::ffff:127.0.0.1 | User GUID: b225cb76cfd6486d85da90ec5b775f2d@bceidboth, Event: Logged Out, Portal: Institution.",
+    );
+  });
+
+  it(`Should log 'Session Timed Out' message when ${AuditEvent.SessionTimedOut} is requested.`, async () => {
+    // Arrange
+    const endpoint = `/audit/${AuditEvent.SessionTimedOut}`;
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED);
+
+    expect(auditService.logger.log).toHaveBeenCalledWith(
+      "SIMS Audit Event From ::ffff:127.0.0.1 | User GUID: student_e2e_test, Event: Session Timed Out, Portal: Student.",
+    );
+  });
+
+  it(`Should log 'Browser Closed' message when ${AuditEvent.BrowserClosed} is requested.`, async () => {
+    // Arrange
+    const endpoint = `/audit/${AuditEvent.BrowserClosed}`;
+    const token = await getAESTToken(AESTGroups.Operations);
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED);
+
+    expect(auditService.logger.log).toHaveBeenCalledWith(
+      "SIMS Audit Event From ::ffff:127.0.0.1 | User GUID: ministry-user-aest-operations@e2e-tests, Event: Browser Closed, Portal: Ministry.",
+    );
+  });
+
+  it(`Should log 'Browser Reopened' message when ${AuditEvent.BrowserReopened} is requested.`, async () => {
+    // Arrange
+    const endpoint = `/audit/${AuditEvent.BrowserReopened}`;
+    const token = await getAESTToken(AESTGroups.Operations);
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED);
+
+    expect(auditService.logger.log).toHaveBeenCalledWith(
+      "SIMS Audit Event From ::ffff:127.0.0.1 | User GUID: ministry-user-aest-operations@e2e-tests, Event: Browser Reopened, Portal: Ministry.",
+    );
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
```

Comment:
So far, I believe that every E2E uses the `beforeEach` at the top. Can we follow the same?

Reply:
Nope, I meant `beforeEach`.
If searched by `beforeEach` in the codebase there will be 37 results and none for `afterEach`. 
Both would work in this case, but we have been using the first one as far as I can see.